### PR TITLE
RDM-12961: supplementary

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-12961
https://tools.hmcts.net/jira/browse/RDM-12963

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java [hunk 1]
```diff
@@ -32,19 +35,24 @@
     private final CaseDefinitionRepository caseDefinitionRepository;
     private final AccessControlService accessControlService;
     private final CaseAccessService caseAccessService;
-
+    private SupplementaryDataUpdateOperation supplementaryDataUpdateOperation;
+    private SupplementaryDataUpdateRequestValidator validator;
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java:39]:
```suggestion
    private SupplementaryDataUpdateRequestValidator supplementaryDataValidator;
```

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java [hunk 2]
```diff
@@ -32,19 +35,24 @@
...
 
     public AuthorisedCreateCaseOperation(@Qualifier("classified") final CreateCaseOperation createCaseOperation,
                                          @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
                                          final CaseDefinitionRepository caseDefinitionRepository,
                                          final AccessControlService accessControlService,
-                                         final CaseAccessService caseAccessService) {
+                                         final CaseAccessService caseAccessService,
+                                         @Qualifier("authorised")
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java:46]: I think we could probably just use the `default` one? Then move this code into the `DefaultCreateCaseOperation`. As it's part of the create case, the user must already have access to the case so no need to go to `AuthorisedSupplementaryDataUpdateOperation` and get the case details again to verify if they have access?

**Review by danlysiak — COMMENTED** [review]
Couple of minor comments, otherwise looks good 👍

File: src/aat/resources/features/F-130 - Create Case Citizen V1 Supplementary Data/F130_Case_Data_Create_Token_Creation.td.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+	"_guid_": "F130_Case_Data_Create_Token_Creation",
+	"_extends_": "Befta_Jurisdiction2_Default_Token_Creation_Data_For_Case_Creation_Solicitor_1",
+	"specs": [
+		"to create a token for case creation"
+	],
+	"users": {
+		"invokingUser": {
+			"username": "befta.caseworker.2.solicitor.2@gmail.com",
```
[the remark this answers is not in the document]
Comment (kiran-yenigala-hmcts) [reply src/aat/resources/features/F-130 - Create Case Citizen V1 Supplementary Data/F130_Case_Data_Create_Token_Creation.td.json:9]: seems like we don't have env variable defined for this user name !! will keep this for now

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java:46]: done the change

**Review by danlysiak — APPROVED** [review]
✅

**Pull request merged**